src/main/scala/sigmastate/utxo/ErgoContext.scala

```diff
@@ -52,34 +52,40 @@ object ErgoContext {
 case object Height extends NotReadyValueInt {
   override val opCode: OpCode = OpCodes.HeightCode
 
-  override lazy val cost: Int = Cost.HeightAccess
+  override def cost[C <: Context[C]](context: C): Long = 2 * Cost.IntConstantDeclaration
 }
 
 /** When interpreted evaluates to a collection of BoxConstant built from Context.boxesToSpend */
 case object Inputs extends LazyCollection[SBox.type] {
   override val opCode: OpCode = OpCodes.InputsCode
 
-  val cost = 1
+  override def cost[C <: Context[C]](context: C) =
+    context.asInstanceOf[ErgoContext].boxesToSpend.size * Cost.BoxConstant
+
```

Comment:
@kushti 
If we assume that Context is immutable during script execution and all boxes in the context already have attached registers then Inputs.cost should include costs of all registers.
In other words we need to measure the cost of Inputs and Outputs checking the size of the all data in them.

Reply:
@ergomorphic ok, I've made a new pricing function for a box based on its size